ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -384,31 +384,18 @@ export function WorkflowDetails({history, location, match}: RouteComponentProps<
     // Get workflow
     useEffect(() => {
         (async () => {
-            let archivedWf: Workflow;
-            if (uid !== '') {
-                try {
-                    archivedWf = await services.workflows.getArchived(namespace, uid);
-                    setError(null);
-                } catch (err) {
-                    if (err.status !== 404) {
-                        setError(err);
-                    }
-                }
-            }
-
             try {
                 const wf = await services.workflows.get(namespace, name);
-                setError(null);
-                // If we find live workflow which has same uid, we use live workflow.
-                if (!archivedWf || archivedWf.metadata.uid === wf.metadata.uid) {
-                    setWorkflow(wf);
-                    setUid(wf.metadata.uid);
-                } else {
-                    setWorkflow(archivedWf);
-                }
+                setUid(wf.metadata.uid);
+                setWorkflow(wf);
             } catch (err) {
-                if (archivedWf) {
-                    setWorkflow(archivedWf);
+                if (err.status === 404 && uid !== '') {
```

Comment:
you can invert this conditional and return early to reduce nesting

Reply:
Done.